File: hfsfolder-ironruby/model/PreFile.rb
# 
# * Created by SharpDevelop.
# * User: henrique.souza
# * Date: 08/04/2019
# * Time: 14:00
# *
# * To change this template use Tools | Options | Coding | Edit Standard Headers.
# 
require "mscorlib"
require "System, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"

module hfsfolder.model
	# <summary>
	# Description of PreFile.
	# </summary>
	class PreFile
		def initialize(name, size, modified, attributes, formatSize, formatModified)
			self.@name = name
			self.@size = size
			self.@modified = modified
			self.@attributes = attributes
			self.@formatedSize = formatSize
			self.@formatedModified = formatModified
		end

		def initialize(name, size, modified, attributes, formatSize, formatModified)
			self.@name = name
			self.@size = size
			self.@modified = modified
			self.@attributes = attributes
			self.@formatedSize = formatSize
			self.@formatedModified = formatModified
		end

		def getName()
			return @name
		end

		def setName(name)
			self.@name = name
		end

		def getSize()
			return @size
		end

		def setSize(size)
			self.@size = size
		end

		def getModified()
			return @modified
		end

		def setModified(modified)
			self.@modified = modified
		end

		def getAttributes()
			return @attributes
		end

		def setAttributes(attributes)
			self.@attributes = attributes
		end

		def getFormatedSize()
			return @formatedSize
		end

		def setFormatedSize(formatedSize)
			self.@formatedSize = formatedSize
		end

		def getFormatedModified()
			return @formatedModified
		end

		def setFormatedModified(formatedModified)
			self.@formatedModified = formatedModified
		end

		def getOriginalPath()
			return @originalPath
		end

		def setOriginalPath(originalPath)
			self.@originalPath = originalPath
		end

		def isDirectory()
			return @directory
		end

		def setDirectory(directory)
			self.@directory = directory
		end

		def limparDados()
			self.@name = ""
			self.@size = 0
			self.@modified = DateTime.Now
			self.@attributes = ""
			self.@formatedSize = ""
			self.@formatedModified = ""
			self.@originalPath = ""
			self.@directory = false
		end

		def ToString()
			return "PreFile [name=" + @name + ", size=" + @size + ", modified=" + @modified + ", attributes=" + @attributes + ", formatSize=" + @formatedSize + ", formatModified=" + @formatedModified + "]"
		end

		def ToInsert()
			return "'" + self.getName() + "'," + @size + ",'" + @formatedModified + "','" + @attributes + "'"
		end

		def ToCVS()
			return self.getName() + ";" + @size + ";" + @formatedModified + ";" + @attributes
		end
	end
end

File: hfsfolder-ironpython/model/PreFile.py
# 
# * Created by SharpDevelop.
# * User: henrique.souza
# * Date: 08/04/2019
# * Time: 14:00
# *
# * To change this template use Tools | Options | Coding | Edit Standard Headers.
# 
from System import *

class PreFile(object):
	""" <summary>
	 Description of PreFile.
	 </summary>
	"""
	def __init__(self, name, size, modified, attributes, formatSize, formatModified):
		self._name = name
		self._size = size
		self._modified = modified
		self._attributes = attributes
		self._formatedSize = formatSize
		self._formatedModified = formatModified

	def __init__(self, name, size, modified, attributes, formatSize, formatModified):
		self._name = name
		self._size = size
		self._modified = modified
		self._attributes = attributes
		self._formatedSize = formatSize
		self._formatedModified = formatModified

	def getName(self):
		return self._name

	def setName(self, name):
		self._name = name

	def getSize(self):
		return self._size

	def setSize(self, size):
		self._size = size

	def getModified(self):
		return self._modified

	def setModified(self, modified):
		self._modified = modified

	def getAttributes(self):
		return self._attributes

	def setAttributes(self, attributes):
		self._attributes = attributes

	def getFormatedSize(self):
		return self._formatedSize

	def setFormatedSize(self, formatedSize):
		self._formatedSize = formatedSize

	def getFormatedModified(self):
		return self._formatedModified

	def setFormatedModified(self, formatedModified):
		self._formatedModified = formatedModified

	def getOriginalPath(self):
		return self._originalPath

	def setOriginalPath(self, originalPath):
		self._originalPath = originalPath

	def isDirectory(self):
		return self._directory

	def setDirectory(self, directory):
		self._directory = directory

	def limparDados(self):
		self._name = ""
		self._size = 0
		self._modified = DateTime.Now
		self._attributes = ""
		self._formatedSize = ""
		self._formatedModified = ""
		self._originalPath = ""
		self._directory = False

	def ToString(self):
		return "PreFile [name=" + self._name + ", size=" + self._size + ", modified=" + self._modified + ", attributes=" + self._attributes + ", formatSize=" + self._formatedSize + ", formatModified=" + self._formatedModified + "]"

	def ToInsert(self):
		return "'" + self.getName() + "'," + self._size + ",'" + self._formatedModified + "','" + self._attributes + "'"

	def ToCVS(self):
		return self.getName() + ";" + self._size + ";" + self._formatedModified + ";" + self._attributes